simplelru/lru.go

```diff
@@ -12,31 +12,52 @@ import (
 // EvictCallback is used to get a callback when a cache entry is evicted
 type EvictCallback[K comparable, V any] func(key K, value V)
 
-// LRU implements a non-thread safe fixed size LRU cache
-type LRU[K comparable, V any] struct {
+// Cache implements a non-thread safe fixed cache with Cache and SIEVE eviction (https://cachemon.github.io/SIEVE-website/)
+type Cache[K comparable, V any] struct {
```

Comment:
This rename doesn't look like something you want to do.  Yes, I know that this makes more sense (lru.Cache vs lru.LRU)...  but you are forcing a Major version change.  The naming here isn't that important, even if annoying.  

I know with the advent of semantic versioning we have a lot of people making version 100 of something, but I think that is bad practice without major benefits.

I'd also package up a name change as its own PR if everyone agrees this is necessary for some reason.  But I like err on the "don't break userspace" guideline and this is breaking compatibility I think without a big enough reason. 

Also, the comment shouldn't be changed.  Yes, this has Cache and SIEVE eviction, but that shouldn't be noted here.  Those are impelmentation details of a cache and is documented in the constructor.